x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/EsqlParser.java

```diff
@@ -32,6 +32,16 @@ public class EsqlParser {
 
     private static final Logger log = LogManager.getLogger(EsqlParser.class);
 
+    private EsqlConfig config = new EsqlConfig();
```

Comment:
It might however for testing purposes I think it's better to not do it this way. At runtime we'd have only one ES cluster and thus the cluster settings would be the same across the parser. However each query would create its own parser instance and having it's configuration, is overall better.
We could use the same EsqlConfig instance across all of them (created through the EsqlPlugin) if we wanted to.